client/datasync-client/src/DataManager.ts

```diff
@@ -18,77 +19,92 @@ export class DataManager {
 
   /**
    * Listing all data for this data set.
+   *
+   * @returns Promise (bluebird)
    */
-  public list(callback: (err?: Error, results?: any) => void) {
+  public list(): Bluebird<any> {
```

Comment:
I think this could at least be an array type instead of bare `any`, helps client code a little.

Reply:
To be clear - you mean `any[]` ?

Reply:
Yes I think that works